packages/app/src/search/actions/search.action.ts

```diff
@@ -8,10 +8,16 @@ import { SearchEvent } from "App/search/constants"
 import { SearchParams, SearchResult } from "App/search/dto"
 import { searchRequest } from "App/search/requests"
 
-export const search = createAsyncThunk<SearchResult | undefined, SearchParams>(
-  SearchEvent.SearchData,
-  async (payload) => {
-    const result = await searchRequest(payload)
-    return result.ok ? result.data : undefined
-  }
+const searchActionGenerator = (searchAction: SearchEvent) =>
```

Comment:
🟢 
A nice abstract to save a boilerplate. However, I'm not sure that is a good use case according to a DRY principle. 

> This is all well and good until you realize that now you’ve taken something simple and turned it into something really complicated that even you have a hard time understanding anymore.

source: [Balancing DRY and readability](http://jonkruger.com/blog/2014/02/24/balancing-dry-and-readability/) or something other [dry vs readability](https://www.google.com/search?q=dry+vs+readability&oq=DRY+vs+read&aqs=chrome.1.69i57j0i22i30l2j0i390l5.9191j0j7&sourceid=chrome&ie=UTF-8)

In general for this project... we don't create action generators to avoid mismatch and allow for easy maintenance of action logic when maintenance requires splitting or making it in another way 😉 However, I am fully open to your perspective if you have a different opinion 😇 🍀

Reply:
I totally agree that readability is more important than DRY. However in this case I have a feeling that action generator improves readability. Here's why:
- only actions are exported so abstraction have no impact readability in other files
- generator itself is really short and not too hard to understand, it basically passing one parameter (in other words it's not decreasing readability too much)
- and the most important point. it gives clear information that each of search actions do exactly the same, and only thing that differs those actions is name (which improves readability)

Regarding maintenance. In case of split action logic I agree that there is additional work to do, but if in future we want to move filtering/searching logic from other components to actions we can use same generator.

Reply:
Sure! I'm not going to convince you. I understand that we may have different opinions on this matter and I respect your perspective 🙇